src/evaluate/harness/__init__.py

```diff
@@ -0,0 +1,71 @@
+import json
+import os
+
+from evaluate import evaluator
+from evaluate.loading import convert_hf_hub_path, relative_to_absolute_path
+from evaluate.utils.file_utils import cached_path, is_relative_path
+from evaluate.utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class Harness:
+    """
+    This class instantiates an evaluation harness made up of multiple tasks, where each task consists of a dataset and
+    an associated metric, and runs evaluation on a model or pipeline. Harnesses can be instantiated from a JSON file
+    named harness_config.json found either locally or uploaded as a dataset on the Hugging Face Hub.
+
+    Usage:
+    ```python
+    >>> from evaluate.harness import Harness
+
+    >>> harness = Harness('mathemakitten/glue-harness')
+    >>> results = harness.run(model_or_pipeline='gpt2')
+    ```
+    """
+
+    def __init__(self, path):
+        """Instantiates a Harness object from a JSON file which will be passed to the Evaluator to run evaluation for a
+        model on this collection of tasks."""
+
+        filename = list(filter(lambda x: x, path.replace(os.sep, "/").split("/")))[-1]
+        if not filename.endswith(".json"):
+            filename = filename + ".json"
+        if path.endswith(filename):  # Try locally
+            if os.path.isfile(path):
+                json_filepath = path
+                self.config = json.load(open(json_filepath))
+            else:
+                raise FileNotFoundError(f"Couldn't find a configuration file at {relative_to_absolute_path(path)}")
+        # Load from a dataset on the Hub
+        elif is_relative_path(path) and path.count("/") <= 1:
+            json_filepath = cached_path(convert_hf_hub_path(os.path.join(path, "harness_config.json")))
```

Comment:
In the evaluation modules we have the convention that the script is in `module_name/module_name.py`. Maybe we can do something similar for harnesses.